### PR TITLE
Fix username clipping in sidebar

### LINKED ISSUE #4000: Profile: side nav menu
**Describe the bug**
When the users name is long, it gets cut at the edge of the side nav bar
Screenshot 2022-11-08 at 1 06 04 PM

Make the users name fix properly in the space

### PR DESCRIPTION
**Pull request by Ashesh3**
Fixes #4000

image

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

### CODE REVIEW

File: src/Components/Common/Sidebar/SidebarUserCard.tsx [hunk 1]
```diff
@@ -22,10 +22,10 @@ const SidebarUserCard = ({ shrinked }: { shrinked: boolean }) => {
         <i className="text-white text-3xl uil uil-user-circle" />
       </Link>
       <div className={`${shrinked ? "hidden" : "grow"} pl-3 flex flex-col`}>
-        <div className="h-6 flex items-center">
+        <div className="min-h-6 flex items-center">
           <Link
             href="/user/profile"
-            className="font-semibold text-white flex-nowrap overflow-hidden truncate"
+            className="font-semibold text-white flex-nowrap overflow-hidden break-all"
```
Comment (rithviknishad) [inline src/Components/Common/Sidebar/SidebarUserCard.tsx]: How about breaking words instead?
```suggestion
            className="font-semibold text-white flex-nowrap overflow-hidden break-words"
```

Comment (Ashesh3) [reply src/Components/Common/Sidebar/SidebarUserCard.tsx]: Fails in some cases where the name contains a single word that is too long to fit.

`break-words`
image

`break-all`
image

(Fun fact: It's a real name!)

Comment (rithviknishad) [reply src/Components/Common/Sidebar/SidebarUserCard.tsx]: the width is probably not correct then. it should ideally work if the width is correct. (https://tailwindcss.com/docs/word-break#break-words)
image

Comment (rithviknishad) [reply src/Components/Common/Sidebar/SidebarUserCard.tsx]: maybe that's why even `truncate` didn't work earlier

Comment (Ashesh3) [reply src/Components/Common/Sidebar/SidebarUserCard.tsx]: Let me take a look at this, ~~by the way what are the benefits of doing this over `break-all`~~?

Comment (Ashesh3) [reply src/Components/Common/Sidebar/SidebarUserCard.tsx]: Got it working with `break-words`, Reference: https://codepen.io/lehollandaisvolant/pen/MvOYxp
image

**Comment by nihal467** [comment]
test approved

**Review by rithviknishad — APPROVED** [review]

**Pull request merged**